src/backend/Device.h

```diff
@@ -52,6 +52,8 @@ namespace backend {
             virtual TextureViewBase* CreateTextureView(TextureViewBuilder* builder) = 0;
 
             virtual void TickImpl() = 0;
+            virtual void ReferenceImpl() = 0;
```

Comment:
nit: I don't think we'll ever need to do something inside these Impl() so we could remove them